examples/bigbench_hard/bigbench_hard.py

```diff
@@ -0,0 +1,361 @@
+import os
+import re
+import logging
+import datetime
+import json
+from pathlib import Path
+from typing import Dict, List, Callable, Union, Any
+
+from graph_of_thoughts.operations import Thought
+from graph_of_thoughts import controller, language_models, operations, prompter, parser
+import project_utils as project
+from bbh_tasks import BBHTask as BBH_Tasks
+
+
+def extract_answer(text: str):
+    match = re.search(r'<Answer>(.*?)<\/Answer>', text)
+    if match:
+        return match.group(1)
+    return None
+
+
+def score_answers_by_frequency(thoughts: List[Thought]) -> List[Thought]:
+    ignore_empty_answers = False  # Setting to true helps with Llama3 as it often doesn't give answer in parseable form.
+    scores = {}
+    for thought in thoughts:
+        current_state = thought.state["current"]
+        scores[current_state] = scores.get(current_state, 0) + 1
+    logging.info("return_most_frequent_answer: scores: {}".format(scores))
+    frequent_answer = max(scores, key=scores.get)
+    for thought in thoughts:
+        thought.score = scores[thought.state["current"]]
+        if ignore_empty_answers:
+            if thought.state is None:
+                thought.score = 0
+    return thoughts
```

Comment:
smart to score by frequency! I just did 1 if majority else 0. Will add this too (or we could make this scoring function part of the score operation in an extra branch.. I think we will need it more often..)

Reply:
Yes, Let's implement a `RelativeScorer` in our next iteration